### PR TITLE
AIT environment improvements

### PR DESCRIPTION
**Pull request by GGP1**
|Related issue|
|---|
|Closes https://github.com/wazuh/wazuh/issues/22891 |

## Description

Introduces changes to the API integration tests environment to improve its efficiency and reliability.

### Tests

Integration burst: [artifacts.zip](https://github.com/wazuh/wazuh/files/15221496/artifacts.zip)

#### Test with a branch that is not on remote (no tarball)

The manager image uses the current branch tarball to build Wazuh, if the local branch is not on remote it should fail early.

<details><summary>Logs</summary>

```console
(venv_connexion) gasti@pop-os:~/work/wazuh/api/test/integration$ pytest -vv test_active_response_endpoints.tavern.yaml --disable-warnings
================================================================= test session starts ==================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.4.0
rootdir: /home/gasti/work/wazuh/api/test/integration
configfile: pytest.ini
plugins: trio-0.8.0, asyncio-0.18.1, tavern-1.23.5, metadata-3.1.1, html-2.1.1, anyio-4.3.0
asyncio: mode=auto
collected 2 items                                                                                                                                      

test_active_response_endpoints.tavern.yaml EEE                                                                                                   [100%]

======================================================================== ERRORS ========================================================================
...
=============================================================== short test summary info ================================================================
ERROR test_active_response_endpoints.tavern.yaml::PUT /active-response/{:agent_id} - Failed: Current branch tarball doesn't exist
ERROR test_active_response_endpoints.tavern.yaml::PUT /active-response - Failed: Current branch tarball doesn't exist
ERROR test_active_response_endpoints.tavern.yaml::PUT /active-response - TypeError: 'NoneType' object is not callable
============================================================= 1 warning, 3 errors in 1.08s =============================================================
```

</details>

#### Test after pushing the changes

docker.log: [docker_log.zip](https://github.com/wazuh/wazuh/files/15169102/docker_log.zip)

<details><summary>Logs</summary>

```console
(venv_connexion) gasti@pop-os:~/work/wazuh/api/test/integration$ pytest -vv test_active_response_endpoints.tavern.yaml --disable-warnings
================================================================= test session starts ==================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.4.0 -- /home/gasti/work/wazuh/venv_connexion/bin/python
cachedir: .pytest_cache
metadata: {'Python': '3.10.12', 'Platform': 'Linux-6.6.10-76060610-generic-x86_64-with-glibc2.35', 'Packages': {'pytest': '7.3.1', 'pluggy': '1.4.0'}, 'Plugins': {'trio': '0.8.0', 'asyncio': '0.18.1', 'tavern': '1.23.5', 'metadata': '3.1.1', 'html': '2.1.1', 'anyio': '4.3.0'}}
rootdir: /home/gasti/work/wazuh/api/test/integration
configfile: pytest.ini
plugins: trio-0.8.0, asyncio-0.18.1, tavern-1.23.5, metadata-3.1.1, html-2.1.1, anyio-4.3.0
asyncio: mode=auto
collected 2 items                                                                                                                                      

test_active_response_endpoints.tavern.yaml::PUT /active-response/{:agent_id} PASSED                                                              [ 50%]
test_active_response_endpoints.tavern.yaml::PUT /active-response PASSED                                                                          [100%]

====================================================== 2 passed, 3 warnings in 520.57s (0:08:40) =======================================================

```

</details>

#### Docker images before and after the changes in the docker-compose.yaml

Before the changes we were building one image per node and agent, now we are reusing them.

<details><summary>Before</summary>

```console
gasti@pop-os:~/work/wazuh/api/test/integration$ docker images
REPOSITORY                         TAG       IMAGE ID       CREATED              SIZE
<none>                             <none>    b892328b9b94   About a minute ago   2.28GB
<none>                             <none>    0c861d926de9   About a minute ago   2.28GB
integration_test_wazuh-manager     latest    c4a46c14bda1   About a minute ago   2.28GB
<none>                             <none>    f24083273883   2 minutes ago        1.21GB
<none>                             <none>    73aa0eda5c19   2 minutes ago        1.21GB
<none>                             <none>    075cd8f4d8fe   2 minutes ago        1.21GB
integration_test_wazuh-agent       latest    f30e6ca8f2a9   2 minutes ago        1.21GB
<none>                             <none>    827aeed56891   4 minutes ago        160MB
integration_test_wazuh-agent_old   latest    cab85f426790   4 minutes ago        160MB
<none>                             <none>    4163237b0eff   4 minutes ago        160MB
<none>                             <none>    4affc4c5136f   4 minutes ago        160MB
integration_test_nginx-lb          latest    1cd99f4565c8   5 minutes ago        188MB
```

</details>

<details><summary>After</summary>

```console
gasti@pop-os:~/work/wazuh/api/test/integration/env$ docker images
REPOSITORY                         TAG       IMAGE ID       CREATED          SIZE
integration_test_wazuh-manager     latest    1c3fed9836cc   12 minutes ago   2.28GB
integration_test_wazuh-agent       latest    ee2e738c3fc1   14 minutes ago   1.21GB
integration_test_wazuh-agent_old   latest    e688c0da1885   17 minutes ago   160MB
integration_test_nginx-lb          latest    9847182cbcbb   17 minutes ago   188MB
```

</details>

#### Launching the test with the `--nobuild` flag doesn't recreate the images

docker.log: [docker_log.zip](https://github.com/wazuh/wazuh/files/15169168/docker_log.zip)

<details><summary>Logs</summary>

Window 1

```console
(venv_connexion) gasti@pop-os:~/work/wazuh/api/test/integration$ pytest -vv test_active_response_endpoints.tavern.yaml --disable-warnings --nobuild
================================================================= test session starts ==================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.4.0 -- /home/gasti/work/wazuh/venv_connexion/bin/python
cachedir: .pytest_cache
metadata: {'Python': '3.10.12', 'Platform': 'Linux-6.6.10-76060610-generic-x86_64-with-glibc2.35', 'Packages': {'pytest': '7.3.1', 'pluggy': '1.4.0'}, 'Plugins': {'trio': '0.8.0', 'asyncio': '0.18.1', 'tavern': '1.23.5', 'metadata': '3.1.1', 'html': '2.1.1', 'anyio': '4.3.0'}}
rootdir: /home/gasti/work/wazuh/api/test/integration
configfile: pytest.ini
plugins: trio-0.8.0, asyncio-0.18.1, tavern-1.23.5, metadata-3.1.1, html-2.1.1, anyio-4.3.0
asyncio: mode=auto
collected 2 items                                                                                                                                      

test_active_response_endpoints.tavern.yaml::PUT /active-response/{:agent_id} PASSED                                                              [ 50%]
test_active_response_endpoints.tavern.yaml::PUT /active-response PASSED                                                                          [100%]

====================================================== 2 passed, 3 warnings in 133.44s (0:02:13) =======================================================
```

Window 2

```console
gasti@pop-os:~/work/wazuh/api/test/integration/env$ docker images
REPOSITORY                         TAG       IMAGE ID       CREATED          SIZE
integration_test_wazuh-agent_old   latest    8cd3ea05c80b   13 minutes ago   160MB
integration_test_nginx-lb          latest    5a200b749c23   13 minutes ago   188MB
integration_test_wazuh-manager     latest    1c3fed9836cc   26 minutes ago   2.28GB
integration_test_wazuh-agent       latest    ee2e738c3fc1   28 minutes ago   1.21GB
```

</details>

### CODE REVIEW

File: api/test/integration/env/docker-compose.yml [hunk 1]
```diff
@@ -112,16 +104,13 @@ services:
       - nginx-lb
       - wazuh-agent3
     depends_on:
-      - wazuh-agent2
+      - wazuh-agent1
```
Comment (GGP1) [inline api/test/integration/env/docker-compose.yml:107]: All agents using the `new` image must depend on the agent which is building it (agent1), the same applies for the old image (agent5)

File: api/test/integration/env/docker-compose.yml [hunk 2]
```diff
@@ -226,6 +205,11 @@ services:
     build:
       context: ./base/nginx-lb
     image: integration_test_nginx-lb
+    restart: always
```
Comment (GGP1) [inline api/test/integration/env/docker-compose.yml:208]: I found an edge case in which the nginx service exited because the workers weren't up yet, I added the restart policy and dependencies to avoid this.

File: api/test/integration/conftest.py [hunk 3]
```diff
@@ -403,27 +389,29 @@ def clean_up_env():
         enable_white_mode()
 
     general_procedure(module)
-    values = build_and_up(interval=10, build=request.config.getoption('--nobuild'), env_mode=env_mode)
+    build_and_up(build=request.config.getoption('--nobuild'), env_mode=env_mode)
```
Comment (GGP1) [inline api/test/integration/conftest.py:392]: This was right in the end, the variable type is `store_false` meaning that it's always True unless specified.

File: api/test/integration/test_agent_DELETE_endpoints.tavern.yaml [hunk 4]
```diff
@@ -835,4 +835,5 @@ stages:
             - "006"
             - "007"
             - "008"
-          total_affected_items: 3
+            - "013"
```
Comment (GGP1) [inline api/test/integration/test_agent_DELETE_endpoints.tavern.yaml:838]: Agent 004 consistently gets to reconnect before the test ends

File: api/test/integration/conftest.py [hunk 5]
```diff
@@ -80,90 +80,76 @@ def pytest_tavern_beta_before_every_test_run(test_dict, variables):
...
     else:
         current_branch = ref.split("/")[-1]
-        
-    os.makedirs(test_logs_path, exist_ok=True)
+
+    if build:
+        # Ping the current branch tarball used to build the manager image.
+        response = requests.get(f"https://github.com/wazuh/wazuh/tarball/{current_branch}")
+        if response.status_code == 404:
+            pytest.fail("Current branch tarball doesn't exist")
```
Comment (fdalmaup) [inline api/test/integration/conftest.py:108]: We should include a condition for when the response code is not successful along with a message describing it.

**Review by fdalmaup — CHANGES_REQUESTED** [review]
Given the modifications made to the `conftest.py` and `docker-compose.yml` files, every test should be run to verify the results are as expected.

**Comment by GGP1** [comment]
## Test reports

Some test failed but after running the same build in `4.9.0` I noticed they fail there as well, so it's related to the base branch and not to the changes introduced in this PR. Some of the tests passed when executed locally though.

<details><summary>fix/22891-ait-env</summary>

Build: https://ci.wazuh.info/job/Test_integration_endpoints/4520/
Artifacts: [artifacts.zip](https://github.com/wazuh/wazuh/files/15223013/artifacts.zip)

| **Test name** | **Pass** | **XPass** | **Skip** | **XFail** | **Fail** | **Issues Ref.** | **Status** |
|--|--|--|--|--|--|--|--|
| test_active_response_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_agent_DELETE_endpoints.tavern.yaml | 6 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_agent_GET_endpoints.tavern.yaml | 95 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_agent_POST_endpoints.tavern.yaml | 6 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_agent_PUT_endpoints.tavern.yaml | 10 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_cdb_list_endpoints.tavern.yaml | 5 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_cluster_endpoints.tavern.yaml | 49 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_decoder_endpoints.tavern.yaml | 25 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_default_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_event_endpoints.tavern.yaml | 5 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_experimental_endpoints.tavern.yaml | 12 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_logtest_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_manager_endpoints.tavern.yaml | 35 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_mitre_endpoints.tavern.yaml | 7 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_overview_endpoints.tavern.yaml | 0 | 0 | 0 | 0 | 1 | insert issue link | :red_circle: |
| test_rbac_black_active_response_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_agent_endpoints.tavern.yaml | 41 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_cdb_list_endpoints.tavern.yaml | 5 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_cluster_endpoints.tavern.yaml | 19 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_decoder_endpoints.tavern.yaml | 6 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_event_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_experimental_endpoints.tavern.yaml | 12 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_logtest_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_manager_endpoints.tavern.yaml | 16 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_mitre_endpoints.tavern.yaml | 7 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_overview_endpoints.tavern.yaml | 0 | 0 | 0 | 0 | 1 | insert issue link | :red_circle: |
| test_rbac_black_rootcheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_rule_endpoints.tavern.yaml | 8 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_sca_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_security_endpoints.tavern.yaml | 25 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_syscheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_syscollector_endpoints.tavern.yaml | 9 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_task_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_active_response_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_agent_endpoints.tavern.yaml | 39 | 0 | 0 | 0 | 2 | insert issue link | :red_circle: |
| test_rbac_white_all_endpoints.tavern.yaml | 162 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_cdb_list_endpoints.tavern.yaml | 5 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_cluster_endpoints.tavern.yaml | 19 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_decoder_endpoints.tavern.yaml | 6 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_event_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_experimental_endpoints.tavern.yaml | 12 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_logtest_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_manager_endpoints.tavern.yaml | 16 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_mitre_endpoints.tavern.yaml | 7 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_overview_endpoints.tavern.yaml | 0 | 0 | 0 | 0 | 1 | insert issue link | :red_circle: |
| test_rbac_white_rootcheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_rule_endpoints.tavern.yaml | 8 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_sca_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_security_endpoints.tavern.yaml | 25 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_syscheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_syscollector_endpoints.tavern.yaml | 9 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_task_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rootcheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rule_endpoints.tavern.yaml | 15 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_sca_endpoints.tavern.yaml | 45 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_security_DELETE_endpoints.tavern.yaml | 15 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_security_GET_endpoints.tavern.yaml | 11 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_security_POST_endpoints.tavern.yaml | 8 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_security_PUT_endpoints.tavern.yaml | 9 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_syscheck_endpoints.tavern.yaml | 34 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_syscollector_endpoints.tavern.yaml | 159 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_task_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |

</details>

<details><summary>4.9.0</summary>

Build: https://ci.wazuh.info/job/Test_integration_endpoints/4524/
Artifacts: [artifacts.zip](https://github.com/wazuh/wazuh/files/15223006/artifacts.zip)

| **Test name** | **Pass** | **XPass** | **Skip** | **XFail** | **Fail** | **Issues Ref.** | **Status** |
|--|--|--|--|--|--|--|--|
| test_active_response_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 1 | insert issue link | :red_circle: |
| test_agent_DELETE_endpoints.tavern.yaml | 6 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_agent_GET_endpoints.tavern.yaml | 95 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_agent_POST_endpoints.tavern.yaml | 6 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_agent_PUT_endpoints.tavern.yaml | 10 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_cdb_list_endpoints.tavern.yaml | 5 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_cluster_endpoints.tavern.yaml | 46 | 0 | 0 | 0 | 3 | insert issue link | :red_circle: |
| test_decoder_endpoints.tavern.yaml | 25 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_default_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_event_endpoints.tavern.yaml | 5 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_experimental_endpoints.tavern.yaml | 12 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_logtest_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_manager_endpoints.tavern.yaml | 35 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_mitre_endpoints.tavern.yaml | 7 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_overview_endpoints.tavern.yaml | 0 | 0 | 0 | 0 | 1 | insert issue link | :red_circle: |
| test_rbac_black_active_response_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_agent_endpoints.tavern.yaml | 41 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_cdb_list_endpoints.tavern.yaml | 5 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_cluster_endpoints.tavern.yaml | 19 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_decoder_endpoints.tavern.yaml | 6 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_event_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_experimental_endpoints.tavern.yaml | 12 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_logtest_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_manager_endpoints.tavern.yaml | 16 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_mitre_endpoints.tavern.yaml | 7 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_overview_endpoints.tavern.yaml | 0 | 0 | 0 | 0 | 1 | insert issue link | :red_circle: |
| test_rbac_black_rootcheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_rule_endpoints.tavern.yaml | 8 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_sca_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_security_endpoints.tavern.yaml | 25 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_syscheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_syscollector_endpoints.tavern.yaml | 9 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_black_task_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_active_response_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_agent_endpoints.tavern.yaml | 41 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_all_endpoints.tavern.yaml | 162 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_cdb_list_endpoints.tavern.yaml | 5 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_cluster_endpoints.tavern.yaml | 19 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_decoder_endpoints.tavern.yaml | 6 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_event_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_experimental_endpoints.tavern.yaml | 12 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_logtest_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_manager_endpoints.tavern.yaml | 16 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_mitre_endpoints.tavern.yaml | 7 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_overview_endpoints.tavern.yaml | 0 | 0 | 0 | 0 | 1 | insert issue link | :red_circle: |
| test_rbac_white_rootcheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_rule_endpoints.tavern.yaml | 8 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_sca_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_security_endpoints.tavern.yaml | 25 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_syscheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_syscollector_endpoints.tavern.yaml | 9 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rbac_white_task_endpoints.tavern.yaml | 1 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rootcheck_endpoints.tavern.yaml | 4 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_rule_endpoints.tavern.yaml | 15 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_sca_endpoints.tavern.yaml | 45 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_security_DELETE_endpoints.tavern.yaml | 15 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_security_GET_endpoints.tavern.yaml | 11 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_security_POST_endpoints.tavern.yaml | 8 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_security_PUT_endpoints.tavern.yaml | 9 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_syscheck_endpoints.tavern.yaml | 34 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_syscollector_endpoints.tavern.yaml | 159 | 0 | 0 | 0 | 0 |  | :green_circle: |
| test_task_endpoints.tavern.yaml | 2 | 0 | 0 | 0 | 0 |  | :green_circle: |

</details>

### Failed tests local execution

<details><summary>test_overview_endpoints.tavern.yaml</summary>

```console
(venv) gasti@gasti:~/work/wazuh/api/test/integration$ pytest -vv test_overview_endpoints.tavern.yaml --nobuild
================================================================= test session starts =================================================================
platform linux -- Python 3.10.14, pytest-7.3.1, pluggy-1.5.0 -- /home/gasti/work/wazuh/venv/bin/python
cachedir: .pytest_cache
metadata: {'Python': '3.10.14', 'Platform': 'Linux-6.1.0-20-amd64-x86_64-with-glibc2.36', 'Packages': {'pytest': '7.3.1', 'pluggy': '1.5.0'}, 'Plugins': {'asyncio': '0.18.1', 'tavern': '1.23.5', 'html': '2.1.1', 'metadata': '3.1.1', 'anyio': '4.1.0', 'trio': '0.8.0', 'aiohttp': '1.0.4'}}
rootdir: /home/gasti/work/wazuh/api/test/integration
configfile: pytest.ini
plugins: asyncio-0.18.1, tavern-1.23.5, html-2.1.1, metadata-3.1.1, anyio-4.1.0, trio-0.8.0, aiohttp-1.0.4
asyncio: mode=auto
collected 1 item                                                                                                                                      

test_overview_endpoints.tavern.yaml::GET /overview/agents PASSED                                                                                [100%]

====================================================== 1 passed, 2 warnings in 140.13s (0:02:20) ======================================================
```

</details>

<details><summary>test_rbac_white_overview_endpoints.tavern.yaml</summary>

```console
(venv) gasti@gasti:~/work/wazuh/api/test/integration$ pytest -vv test_rbac_white_overview_endpoints.tavern.yaml --nobuild
================================================================= test session starts =================================================================
platform linux -- Python 3.10.14, pytest-7.3.1, pluggy-1.5.0 -- /home/gasti/work/wazuh/venv/bin/python
cachedir: .pytest_cache
metadata: {'Python': '3.10.14', 'Platform': 'Linux-6.1.0-20-amd64-x86_64-with-glibc2.36', 'Packages': {'pytest': '7.3.1', 'pluggy': '1.5.0'}, 'Plugins': {'asyncio': '0.18.1', 'tavern': '1.23.5', 'html': '2.1.1', 'metadata': '3.1.1', 'anyio': '4.1.0', 'trio': '0.8.0', 'aiohttp': '1.0.4'}}
rootdir: /home/gasti/work/wazuh/api/test/integration
configfile: pytest.ini
plugins: asyncio-0.18.1, tavern-1.23.5, html-2.1.1, metadata-3.1.1, anyio-4.1.0, trio-0.8.0, aiohttp-1.0.4
asyncio: mode=auto
collected 1 item                                                                                                                                      

test_rbac_white_overview_endpoints.tavern.yaml::GET /overview/agents PASSED                                                                     [100%]

====================================================== 1 passed, 2 warnings in 129.38s (0:02:09) ======================================================
```

</details>

<details><summary>test_rbac_black_overview_endpoints.tavern.yaml</summary>

```console
(venv) gasti@gasti:~/work/wazuh/api/test/integration$ pytest -vv test_rbac_black_overview_endpoints.tavern.yaml --nobuild
================================================================= test session starts =================================================================
platform linux -- Python 3.10.14, pytest-7.3.1, pluggy-1.5.0 -- /home/gasti/work/wazuh/venv/bin/python
cachedir: .pytest_cache
metadata: {'Python': '3.10.14', 'Platform': 'Linux-6.1.0-20-amd64-x86_64-with-glibc2.36', 'Packages': {'pytest': '7.3.1', 'pluggy': '1.5.0'}, 'Plugins': {'asyncio': '0.18.1', 'tavern': '1.23.5', 'html': '2.1.1', 'metadata': '3.1.1', 'anyio': '4.1.0', 'trio': '0.8.0', 'aiohttp': '1.0.4'}}
rootdir: /home/gasti/work/wazuh/api/test/integration
configfile: pytest.ini
plugins: asyncio-0.18.1, tavern-1.23.5, html-2.1.1, metadata-3.1.1, anyio-4.1.0, trio-0.8.0, aiohttp-1.0.4
asyncio: mode=auto
collected 1 item                                                                                                                                      

test_rbac_black_overview_endpoints.tavern.yaml::GET /overview/agents PASSED                                                                     [100%]

====================================================== 1 passed, 2 warnings in 110.44s (0:01:50) ======================================================
```

</details>

The only one I could reproduce locally was `test_rbac_white_agent_endpoints.tavern.yaml` but the failure seems to be related to `wazuh-agentd`. The same endpoints are executed in `test_agent_PUT_endpoints.tavern.yaml` and it passed so it's not something related to the environment I would say.

<details><summary>test_rbac_white_agent_endpoints.tavern.yaml</summary>

```console
E   tavern.util.exceptions.TestFailError: Test 'Try to restart a list of agents (Partially allowed, user aware)' failed:
    - Key mismatch: (expected["error"] = '2' (type = <class 'int'>), actual["error"] = '1' (type = <class 'int'>))
    - Error calling validate function '<function healthcheck_agent_restart at 0x7f288d216440>':
        Traceback (most recent call last):
          File "/home/gasti/work/wazuh/venv/lib/python3.10/site-packages/tavern/response/base.py", line 152, in _maybe_run_validate_functions
            vf(response)
          File "/home/gasti/work/wazuh/venv/lib/python3.10/site-packages/tavern/util/extfunctions.py", line 98, in inner
            result = func(response, *args, **kwargs)
          File "/home/gasti/work/wazuh/api/test/integration/tavern_utils.py", line 511, in healthcheck_agent_restart
            check_agentd_started(response, agents_list)
          File "/home/gasti/work/wazuh/api/test/integration/tavern_utils.py", line 465, in check_agentd_started
            raise ProcessLookupError("The wazuh-agentd daemon was not started after requesting the restart")
        ProcessLookupError: The wazuh-agentd daemon was not started after requesting the restart
```

</details>

**Review by fdalmaup — APPROVED** [review]
LGTM

**Review by Selutario — APPROVED** [review]
Good job @GGP1
